R/estimateAlpha.R

```diff
@@ -0,0 +1,245 @@
+                                                        list(...))))
+        }
+    }
+    return(x)
+}
+
+## Helper functions
+
+.get_indices <- function(measure) {
+    switch(measure,
+           "diversity" = c("coverage_diversity", "faith_diversity", 
+                           "fisher_diversity", "gini_simpson_diversity",
+                           "inverse_simpson_diversity",
+                           "log_modulo_skewness_diversity", "shannon_diversity"),
+           "dominance" = c("absolute_dominance",
+                           "dbp_dominance", "core_abundance_dominance",
+                           "gini_dominance", "dmn_dominance", "relative_dominance",
+                           "simpson_lambda_dominance"),
+           "evenness" = c("camargo_evenness", "pielou_evenness", "simpson_evenness",
+                          "evar_evenness", "bulla_evenness"),
+           "richness" = c("ace_richness", "chao1_richness", "hill_richness", 
+                          "observed_richness"))
+}
+
+.alpha_rarefaction <- function(x,
+                               n.iter=1L,
+                               seed=123,
+                               args.sub=list(assay.type="counts",
+                                             min_size=min(colSums(assay(x, "counts")),
+                                                          na.rm = TRUE),
+                                             verbose=FALSE),
+                               FUN=.estimate_diversity,
+                               args.fun=c(index="shannon",
+                                          assay.type="subsampled"),
+                               ...,
+                               name = args.fun$index) {
+    set.seed(seed)
+    # Calculating the mean of the subsampled alpha estimates ans storing them
+    colData(x)[, name] <- lapply(seq(n.iter), function(j){
+        # subsampling the counts from the original tse object
+        x_sub <- do.call(subsampleCounts, args = c(list(x), args.sub))
+        # calculating the diversity indices on the subsampled object
+        # warnings are supressed due to the depricated warning of the alpha
+        # measure functions
+        suppressWarnings(x_sub <- do.call(FUN, args = c(list(x_sub),
+                                                        args.fun,
+                                                        list(...))))
+        # Storing estimate results
+        colData(x_sub)[, args.fun$index, drop=FALSE]
+    }) %>% as.data.frame() %>% rowMeans() %>% as.data.frame()
+    return(x)
+}
+
+.parse_name <- function(index, name, measure) {
+    # parsing name string to use as a column name at colData when storing estimates
+    if (name==index) {
+        # check if suffix of the alpha indices if present at index
+        # otherwise keeping suffix as a name if name not defined by user.
+        if (measure %in% unlist(strsplit(index, "\\_"))) {
+            name = index
+            } else {
```

Comment:
are the indentations ok here? 

also name=index should be name <- index here i think